students/DrewSmith/lessons/lesson02/assignment/code/charges_calc.py

```diff
@@ -0,0 +1,159 @@
+'''
+Returns total price paid for individual rentals
+
+This module reads a JSON file and expects each section to have 
+product_coderental_start, rental_end, price_per_day and units_rented elements.
+'''
+import argparse
+import json
+import datetime
+import math
+import logging
+
+
+
+def load_logging(log_level, logger=None):
+    '''
+    Loads standard logging setup
+
+    :param log_level: logging level from the user
+    (1: Error, 2: Warning, 3: Debug)
+    '''
+    log = "%(asctime)s %(filename)s:%(lineno)-3d %(levelname)s %(message)s"
+    debug_dict = {'1': logging.ERROR, '2': logging.WARNING, '3': logging.DEBUG}
+    if logger is None:
+        logger = logging.getLogger()
+    logger.setLevel(debug_dict.get(log_level, logging.CRITICAL))
+
+    if log_level in debug_dict:        
+        file_name = f'{datetime.datetime.now().strftime("%Y-%m-%d")}.log'
+        formatter = logging.Formatter(log)
+
+        file_handler = logging.FileHandler(file_name)
+        file_handler.setLevel(logging.WARNING)
+        file_handler.setFormatter(formatter)
+
+        console_handler = logging.StreamHandler()
+        console_handler.setLevel(logging.DEBUG)
+        console_handler.setFormatter(formatter)        
+            
+        logger.addHandler(file_handler)
+        logger.addHandler(console_handler)
+
+def parse_cmd_arguments():
+    ''' Processes commandline arguments from the user '''
+    parser = argparse.ArgumentParser(description='Process some integers.')
+    parser.add_argument('-i', '--input', help='input JSON file', required=True)
+    parser.add_argument('-o', '--output', help='ouput JSON file', required=True)
+    parser.add_argument('-d', '--debug', help='ouput JSON file', required=False)
+
+    result_args = parser.parse_args()
+    load_logging(result_args.debug)
+
+    logging.debug(f"Input: {result_args.input}")
+    logging.debug(f"Output: {result_args.output}")
+    logging.debug(f"Debug: {result_args.debug}")
+
+    return result_args
+
+
+def load_rentals_file(filename):
+    '''
+    Loads the source JSON file
+
+    :param filename: file name to load
+    '''
+
+    result = None
+    try:
+        with open(filename) as file:
+            try:
+                result = json.load(file)
+                logging.debug(f"File loaded successfully: {filename}")
```

Comment:
It is good to limit what is in a try block, and there is such a thing as an else block in a try statement, for this very purpose.
```
try:
   result = json.load(file)
except json.decoder.JSONDecodeError as json_decode_error:
   print(f"Error occurred while reading the JSON file")
   logging.error(f"Error loading JSON: {filename}: " + f"{str(json_decode_error)}")
else:
    logging.debug(f"File loaded successfully: {filename}")
```
I like to think of it as "else if nothing goes wrong"